config/views/sitemaps.py

```diff
@@ -0,0 +1,124 @@
+        for court in courts.get_listable_courts():
+            for year in range(court.start_year, court.end_year):
+                context["maps"].append(
+                    self.request.build_absolute_uri(
+                        reverse("sitemap_court", kwargs={"code": court.canonical_param, "year": year})
+                    )
+                )
+
+        return context
+
+
+class SitemapStaticView(TemplateView, TemplateResponseMixin):
+    content_type = "application/xml"
+    template_name = "sitemaps/sitemap.xml"
+
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        # This is a list of the names of URLs to reverse and include in the map
+        url_names = [
+            "home",
+            "about_this_service",
+            "what_to_expect",
+            "how_to_use_this_service",
+            "courts_and_tribunals",
+            "computational_licence_form",
+            "privacy_notice",
+            "accessibility_statement",
+            "open_justice_licence",
+            "terms_of_use",
+            "publishing_policy",
+            "terms_and_policies",
+            "contact_us",
+            "courts_and_tribunals_in_fcl",
+            "help_and_guidance",
+            "how_to_search_find_case_law",
+            "understanding_judgments_and_decisions",
+            "the_find_case_law_api",
+        ]
+
+        context["items"] = [{"url": self.request.build_absolute_uri(reverse(url))} for url in url_names]
+        return context
+
+
+class SitemapCourtsView(TemplateView, TemplateResponseMixin):
+    content_type = "application/xml"
+    template_name = "sitemaps/sitemap.xml"
+
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        context["items"] = [
+            {
+                "url": self.request.build_absolute_uri(
+                    reverse("court_or_tribunal", kwargs={"param": court.canonical_param})
+                )
+            }
+            for court in courts.get_listable_courts()
+        ]
+
+        return context
+
+
+class SitemapCourtView(TemplateView, TemplateResponseMixin):
+    content_type = "application/xml"
+    template_name = "sitemaps/sitemap.xml"
+
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        court_query = self.kwargs.get("code")
+        year: Union[int, None] = self.kwargs.get("year")
+
+        try:
+            search_parameters = SearchParameters(
+                court=court_query if court_query else None,
+                date_from=(datetime.date(year=year, month=1, day=1).strftime("%Y-%m-%d") if year else None),
+                date_to=(datetime.date(year=year, month=12, day=31).strftime("%Y-%m-%d") if year else None),
+                order="-date",
+                page=1,
+                page_size=50000,
```

Comment:
Is the search API okay handling this? 😄 also are we doing this to avoid there being a chance of 2 pages? I feel like we have more than 50,000 records?

Reply:
Yes; the intersection of court and year should never generate more than a few thousand records at worst (even for the busiest court).